requests/models.py

```diff
@@ -365,11 +365,12 @@ def prepare_url(self, url, params):
         if not host:
             raise InvalidURL("Invalid URL %r: No host supplied" % url)
 
-        # Only want to apply IDNA to the hostname
-        try:
-            host = idna.encode(host, uts46=True).decode('utf-8')
-        except (UnicodeError, idna.IDNAError):
-            raise InvalidURL('URL has an invalid label.')
+        if not host.startswith('xn--') or not is_ascii(host):
```

Comment:
We need a comment to justify why we're leaving some URLs alone: specifically, because we think they've *already* been IDNA-encoded and don't want to stomp on that logic.

We think this if they begin with the IDNA tag (`xn--`) and if all the bytes within them are ASCII.